lib/iris/tests/integration/test_netcdf.py

```diff
@@ -903,5 +907,533 @@ def test_netcdf_with_no_constraint(self):
+
+
+# Attributes to test, which should be 'data' type by default
+_DATA_TEST__ATTRS = (
+    iris.fileformats.netcdf.saver._CF_DATA_ATTRS
+    + iris.fileformats.netcdf.saver._UKMO_DATA_ATTRS
+)
+
+
+# Define a fixture to parametrise over the 'data-style' test attributes.
+# This just provides a more concise way of writing parametrised tests.
+@pytest.fixture(params=_DATA_TEST__ATTRS)
+def data_attr(request):
+    # N.B. "request" is a standard PyTest fixture
+    return request.param  # Return the name of the attribute to test.
+
+
+class TestLoadSaveAttributes:  # (tests.IrisTest):
+    @staticmethod
+    def _calling_testname():
+        """
+        Search up the callstack for a function named "test_*", and return the name for
+        use as a test identifier.
+
+        Returns
+        -------
+        test_name : str
+            Returns a string, with the initial "test_" removed.
+        """
+        test_name = None
+        stack = inspect.stack()
+        for frame in stack[1:]:
+            full_name = frame[3]
+            if full_name.startswith("test_"):
+                # Return the name with the inital "test_" removed.
+                test_name = full_name.replace("test_", "")
+                break
+        # Search should not fail, unless we were called from an inappropriate place?
+        assert test_name is not None
+        return test_name
+
+    def _testfile_path(self, basename: str) -> str:
+        # Make a filepath in the temporary directory, based on the name of the calling
+        # test method, and the "self.attrname" it sets up.
+        testname = self._calling_testname()
+        # Turn that into a suitable temporary filename
+        ext_name = getattr(self, "testname_extension", "")
+        if ext_name:
+            basename = basename + "_" + ext_name
+        path_str = (
+            f"{self.tmpdir}/nc_attr__{self.attrname}__{testname}_{basename}.nc"
+        )
+        return path_str
+
+    @staticmethod
+    def _default_vars_and_attrvalues(vars_and_attrvalues):
+        # Simple default strategy : turn a simple value into {'var': value}
+        if not isinstance(vars_and_attrvalues, dict):
+            # Treat single non-dict argument as a value for a single variable
+            vars_and_attrvalues = {"var": vars_and_attrvalues}
+        return vars_and_attrvalues
+
+    def _create_testcase_files(
+        self,
+        global_attr_value: Optional[str] = None,
+        vars_and_attrvalues: Union[None, str, dict] = None,
+        globalval_file2: Optional[str] = None,
+        var_values_file2: Union[None, str, dict] = None,
+    ):
+        """
+        Create temporary input netcdf files with specific content.
+
+        A generalised routine for creating a netcdf testfile to test behaviour of a
+        specific attribute (name).
+        Create a temporary input netcdf file (or two) with specific global and
+        variable-local versions of a specific attribute.
```

Comment:
There's a bit of redundancy in these sentences; each sentences mentions creating netcdf files. I also find it a bit unclear. 

Perhaps something like:

> Creates a temporary netcdf test file (or two) with the given global and variable-local attributes. The file(s) are used to test the behaviour of the attribute.

Reply:
:+1: